misoc/targets/kasli.py

```diff
@@ -61,44 +99,56 @@ def __init__(self, platform):
         pll_clk200 = Signal()
         self.specials += [
             Instance("MMCME2_BASE",
-                p_CLKIN1_PERIOD=16.0,
-                i_CLKIN1=self.clk125_div2,
+                # 150mhz not supported anymore
+                p_CLKIN1_PERIOD=si5324_period*2,
+                i_CLKIN1=si5324_div2,
 
-                i_CLKFBIN=mmcm_fb,
-                o_CLKFBOUT=mmcm_fb,
+                i_CLKFBIN=mmcm_fb_in,
+                o_CLKFBOUT=mmcm_fb_out,
                 o_LOCKED=mmcm_locked,
 
-                # VCO @ 1GHz with MULT=16
-                p_CLKFBOUT_MULT_F=14.5, p_DIVCLK_DIVIDE=1,
+                # VCO @ 1GHz with MULT=16 (62.5MHz - Kasli 2.0)
+                # VCO @ 800MHz (50MHz - Kasli 1.0/1.1)
+                p_CLKFBOUT_MULT_F=16, p_DIVCLK_DIVIDE=1,
 
-                # ~125MHz
+                # ~125MHz (or 100MHz)
                 p_CLKOUT0_DIVIDE_F=8.0, p_CLKOUT0_PHASE=0.0, o_CLKOUT0=mmcm_sys,
 
-                # ~500MHz. Must be more than 400MHz as per DDR3 specs.
+                # ~500MHz (or 400MHz). Must be more than 400MHz as per DDR3 specs.
                 p_CLKOUT1_DIVIDE=2, p_CLKOUT1_PHASE=0.0, o_CLKOUT1=mmcm_sys4x,
                 p_CLKOUT2_DIVIDE=2, p_CLKOUT2_PHASE=90.0, o_CLKOUT2=mmcm_sys4x_dqs,
             ),
             Instance("PLLE2_BASE",
-                p_CLKIN1_PERIOD=16.0,
-                i_CLKIN1=self.clk125_div2,
+                p_CLKIN1_PERIOD=si5324_period*2,
```

Comment:
Why not just keep that one on clk125?
The IDELAYCTRL 200MHz clock is asynchronous to the others.